lib/deimos/utils/db_poller/base.rb

```diff
@@ -21,17 +21,37 @@ class Base
         # @return [Hash]
         attr_reader :config
 
+        # Method to define producers if a single poller needs to publish to multiple topics.
+        # Producer classes should be constantized
+        # @return [Array<Producer>]
+        def self.producers
+          []
+        end
+
         # @param config [FigTree::ConfigStruct]
         def initialize(config)
           @config = config
           @id = SecureRandom.hex
           begin
-            @producer = @config.producer_class.constantize
-          rescue NameError
-            raise "Class #{@config.producer_class} not found!"
-          end
-          unless @producer < Deimos::ActiveRecordProducer
-            raise "Class #{@producer.class.name} is not an ActiveRecordProducer!"
+            if @config.poller_class.nil? && @config.producer_class.nil?
+              raise 'No producers have been set for this DB poller!'
+            end
+
+            @resource_class = @config.producer_class.present? ? @config.producer_class.constantize : @config.poller_class.constantize
+
+            if @config.poller_class.nil?
+              unless @resource_class < Deimos::ActiveRecordProducer
+                raise "Class #{@resource_class.class.name} is not an ActiveRecordProducer!"
+              end
+            else
+              @resource_class.producers.each do |producer|
+                unless producer < Deimos::ActiveRecordProducer
+                  raise "Class #{producer.class.name} is not an ActiveRecordProducer!"
+                end
```

Comment:
This is some copied code - can we extract this to a method? E.g. `validate_producer_class`?